db/migrate/20241103224635_create_user_roles.rb

```diff
@@ -0,0 +1,28 @@
+class CreateUserRoles < ActiveRecord::Migration[7.1]
+  def up
+    create_table :roles do |t|
+      t.string :name
+
+      t.timestamps
+    end
+
+    create_table :user_roles do |t|
+      t.belongs_to :user
+      t.belongs_to :role
+
+      t.timestamps
+    end
+
+    Role::VALID_ROLES.values.each do |role|
+      Role.create!(name: role)
+    end
```

Comment:
This data migration seems out of place? 
Is there somewhere else we might put this?

Reply:
It was a decision based on convenience, but there's a couple other places where this could be called from:

- Could've used the [data-migration](https://github.com/ilyakatz/data-migrate) gem to manage specifically such migrations: it creates a folder structure, rake tasks to execute them and generators to create them. As the application grows, this would be the ideal approach.

- Ahotner option was to create a rake task specific for this, it would've been a one time execution and then the code for the task would've been removed after execution. I'm not a fan of this approach coz it requires extra effort to remove the rake task afterwards - tho when the data manipulation is complicated, the task is kept as a registry of what was done.